networking/aws_load_balancer_operator/adding-tls-termination-and-ddos-protection.adoc

```diff
@@ -0,0 +1,69 @@
+// Module included in the following assemblies:
+//
+// * networking/aws_load_balancer_operator/add-tls-termination-and-ddos-protection.adoc
+
+:_content-type: PROCEDURE
+[id="nw-adding-tls-termination-and-ddos-protection_{context}"]
+= Adding the tls termination and ddos-protection
+
+You can route traffic for the domain example.com to pods of Service example-service and add TLS termination and DDoS protection on the load balancer.
+
+.Prerequisites
+
+.Procedure
+
+. Deploy the operator and create an instance of the resource `AWSLoadBalancerController`:
++
+[source,yaml]
+----
+kind: AWSLoadBalancerController
+group: networking.olm.openshift.io/v1alpha1 <1>
+metadata:
+  name: cluster <2>
+spec:
+  subnetTagging: auto <3>
+  ingressClass: tls-termination <4>
+----
+
+<1> Defines the resource group of the `AWSLoadBalancerControlller` resource.
+<2> Defines name of a cluster.
+<3> defines subnet tagging performance.
+<4> Defines the name of the default `IngressClass` that the provisioned `aws-load-balancer-controller` reconciles.
+
+. Create an `Ingress` resource:
++
+[source,yaml]
+----
+apiVersion: networking.k8s.io/v1
+kind: Ingress <1>
+metadata:
+  name: example <2>
+  annotations:
+    alb.ingress.kubernetes.io/scheme: internet-facing <3>
+    alb.ingress.kubernetes.io/certificate-arn: arn:aws:acm:us-west-2:xxxxx <4>
+    alb.ingress.kubernetes.io/shield-advanced-protection: true <5>
+spec:
+  ingressClassName: tls-termination <6>
+  rules:
+  - host: example.com <7>
+    http:
+        paths:
+          - path: /
+            pathType: Exact <8>
+            backend:
+              service:
+                name: example-service <9>
+                port:
+                  number: 80 <10>
+----
+
+<1>
+<2>
+<3>
```

Comment:
This annotation makes the controller provision the load balancer for this ingress in a public subnet. So the load balancer is reachable over the internet.

Reply:
Addressed